### PR TITLE
Add E2E tests to disabled welcome modal

### LINKED ISSUE #32157: Setup experiment: disable welcome modals
**User Story:**
As a new merchant
I want to explore my new store without interruptions
So I can start selling as quickly as possible

**Description:**
Most of the information currently displayed in the 3-step modal isn't relevant to merchants who're just starting out. With the changes we're trying out in this cycle, we can replace it with a personalized welcome experience in the Home screen.

**Acceptance Criteria:**

- [ ] Merchants no longer see the welcome modal when they enter the Home screen for the first time

### PR DESCRIPTION
**Pull request by octaedro**
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

This PR adds E2E tests to check that the welcome modal is not being shown.
The welcome modal disabling is part of an experiment, more info [here](https://github.com/woocommerce/woocommerce/pull/32331).

Closes #32157.

### How to test the changes in this Pull Request:

1. Run E2E tests and verify that everything is working correctly.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Add E2E tests to disabled welcome modal

### FOR PR REVIEWER ONLY:

* [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: packages/js/admin-e2e-tests/src/specs/homescreen/welcome-modal.ts
Comment (louwie17) [inline packages/js/admin-e2e-tests/src/specs/homescreen/welcome-modal.ts]: I am not entire opposed to having a whole separate test condition for the welcome modal, but I wonder if something like this should be part of the `complete-onboarding-wizard` spec, given it is specific to onboarding?

If not, I would suggest changing the naming of this test to be more generic and Homescreen focused. `describe('Homescreen` (this would also mean updating the filename.
That way we could add more home screen specific specs in here, like checking if `notes` are rendered for example and if the Stats overview is rendered.

**Review by louwie17 — COMMENTED** [review]
Thanks for adding a specific test case for removing the Welcome modal @octaedro, this tested well :)

I did leave a question below about moving this test to the `complete-onboarding-wizard` spec, as a test case when finishing the onboarding (as it should only show up during the first part of onboarding).
Or
Making the `welcome-modal.ts` more generic, changing it to `homescreen` so we could add future `homescreen` related checks to it.

File: packages/js/admin-e2e-tests/src/specs/homescreen/welcome-modal.ts
Comment (octaedro) [reply packages/js/admin-e2e-tests/src/specs/homescreen/welcome-modal.ts]: Good idea @louwie17! I moved the test to the file `complete-onboarding-wizard` and renamed the test description from "Homescreen welcome modal" to just "Homescreen" in the commit 35a2eb1

**Comment by octaedro** [comment]
Thank you @louwie17 for the review. I addressed the change you suggested.
Could you give another look at this PR?

File: packages/js/admin-e2e-tests/src/specs/activate-and-setup/complete-onboarding-wizard.ts
Comment (louwie17) [inline packages/js/admin-e2e-tests/src/specs/activate-and-setup/complete-onboarding-wizard.ts]: Could you rename this to just `testAdminHomescreen` ?

File: plugins/woocommerce/tests/e2e/specs/admin-homescreen/welcome-modal.test.js
Comment (louwie17) [inline plugins/woocommerce/tests/e2e/specs/admin-homescreen/welcome-modal.test.js]: Could this be removed now, given you are calling it in the `complete-onboarding-wizard.test.js` now?

File: plugins/woocommerce/changelog/add-32157_tests_to_disable_welcome_modals
Comment (louwie17) [inline plugins/woocommerce/changelog/add-32157_tests_to_disable_welcome_modals:4]: We will probably also need a changelog in the `admin-e2e-tests` package, although I don't think we have been very strict with that.

**Review by louwie17 — COMMENTED** [review]
Thanks for making these updates @octaedro, those changes look good, I just left a couple more suggestions for name updates.

File: plugins/woocommerce/changelog/add-32157_tests_to_disable_welcome_modals
Comment (octaedro) [reply plugins/woocommerce/changelog/add-32157_tests_to_disable_welcome_modals:4]: Sounds good to me! I added this change to the changelog in commit 23b3250.

File: plugins/woocommerce/tests/e2e/specs/admin-homescreen/welcome-modal.test.js
Comment (octaedro) [reply plugins/woocommerce/tests/e2e/specs/admin-homescreen/welcome-modal.test.js]: Nice catch! I addressed this change in commit d6c216f

File: packages/js/admin-e2e-tests/src/specs/activate-and-setup/complete-onboarding-wizard.ts
Comment (octaedro) [reply packages/js/admin-e2e-tests/src/specs/activate-and-setup/complete-onboarding-wizard.ts]: Good idea! I renamed the test in commit e219a19

**Comment by octaedro** [comment]
Hey @louwie17,
I just made the modifications you mentioned. Could you take another look at this PR?

**Review by louwie17 — DISMISSED** [review]
LGTM 🚀 Thanks for addressing the feedback @octaedro, this looks good, thanks again for adding extra E2E tests :)

**Review by louwie17 — APPROVED** [review]
Re-approve after merge conflict fix.

**Pull request merged**
